### ISSUE TITLE
CA1801 should ignore unimplemented methods

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn-analyzers#1220]
Given the following code:

```C#
public class C1
{
    public void Unimplemented(int unused)
    {
        throw new NotImplementedException();
    }
}
```

We shouldn't warn about `unused` being unused, as the method isn't implemented.

**Comment by 333fred** [comment]
This is blocked on https://github.com/dotnet/roslyn/issues/8919#issuecomment-305343559.

**Comment by 333fred** [comment]
I've done the implementation for this, we just need to get VB working correctly.

**Comment by mavasani** [comment]
@333fred This is no longer blocked, removing the blocked tag.

**Comment by sharwell** [comment]
> We shouldn't warn about unused being unused, as the method isn't implemented.

@333fred Why? I understand this if the method implements an interface method, but I don't understand in general why we'd want to avoid the warning. Unless there is another diagnostic which reports unimplemented methods separately?

**Comment by 333fred** [comment]
The issue is we don't want to have the ide generate code that is immediately flagged with warnings.

**Comment by sharwell** [comment]
@333Fred I'm not sure I agree it's a problem. Also, since the most prominent cases are implementing an interface, it seems like a small issue. It applies to generating a new method from a call site, but the generated method is a single method that still does need to be implemented.

**Comment by 333fred** [comment]
@mavasani and I talked offline, and we can't come up with any other reasons for this. So I'm going to close this as won't fix and close the pr.

**Comment by sharwell** [comment]
:memo: When I originally read this, I was thinking the warning was reporting that `Unimplemented` (the method) was unused.
